### DISCUSSION TITLE
ESP32: CORRUPT HEAP on send

### DISCUSSION THREAD

**Discussion by raisinbread** [bengtmartensson/Infrared4Arduino#83] [Q&A]
Been having some good luck on Arduino, but needed to move to ESP32. Compiles fine, but when I try to send/emit, I get an error and a crash.

Code:
```
IrSender *sender = IrSenderPwm::getInstance(true);
sender->send(binaryToTxIrSequence(signalData), CARRIER_FREQ, DUTY_CYCLE_PERC);
```

binaryToTxIrSequence() just takes a binary number and creates some spaces and marks for me and returns an IrSequence. When this code gets executed, I end up with this:

```
CORRUPT HEAP: Bad head at 0x3ffb1e6c. Expected 0xabba1234 got 0x3ffc003c
abort() was called at PC 0x400838a1 on core 1

ELF file SHA256: 0000000000000000

Backtrace: 0x40085470:0x3ffb1a50 0x400856e5:0x3ffb1a70 0x400838a1:0x3ffb1a90 0x400839cd:0x3ffb1ac0 0x400d8a8b:0x3ffb1ae0 0x400d4fa5:0x3ffb1da0 0x400d4f40:0x3ffb1df0 0x4008984d:0x3ffb1e20 0x40081af6:0x3ffb1e40 0x40083799:0x3ffb1e60 0x4000bec7:0x3ffb1e80 0x400e7451:0x3ffb1ea0 0x400e73e1:0x3ffb1ec0 0x400d184b:0x3ffb1ee0 0x400d1221:0x3ffb1f00 0x400d137d:0x3ffb1f30 0x400d13d0:0x3ffb1f60 0x400d149d:0x3ffb1f80 0x400d359d:0x3ffb1fb0 0x400866f5:0x3ffb1fd0

Rebooting...
```
I've been using the camera on my phone to see if I can see the IR LED flashing, and it still seems to do so. For some reason it seems like after the IrSequence is sent, things crash. The code works fine on an Uno - wondering why I'm getting crashes here.

New to ESP32 so maybe there's some configuration I'm missing, but wondered if others had seen this or have any clues on how to fix.

**Comment by bengtmartensson** [comment]
Does the example IrSendPwm work?

Please post the entire sketch.

**Reply by raisinbread** [reply]
It does - I think it might be the way I'm constructing the IrSequence. I'll keep playing with it.

**Reply by raisinbread** [reply]
Yes - was an issue with IrSequence creation. Seems to work now.
